data/store.go

```diff
@@ -75,3 +94,39 @@ func (s *Store) GetTransactions() [3]map[string][]DataPoint {
 
 	return transactions
 }
+
+// GetTransactionCounts returns the transaction count per root DID and the total number of roots
+func (s *Store) GetTransactionCounts() (map[string]uint32, uint32) {
+	return s.didCount, s.rootDIDCount
+}
+
+func (s *Store) resolveController(txDID string) (string, bool) {
+	// check if the did is already resolved
+	if controller, ok := s.mapping[txDID]; ok {
+		return controller, false
+	}
+
+	// resolve the did
+	result, err := s.client.DIDDocument(context.Background(), txDID)
+	if err != nil {
+		// resolving failed, just return the original did
+		log.Printf("error resolving did: %s\n", err.Error())
+		return txDID, true
+	}
+
+	root := txDID
+	newRoot := true
+
+	// check if the DID document contains a controller that differs from the did
+	for _, controller := range result.Document.Controller {
+		if controller.String() != txDID {
+			// call resolveController recursively to resolve the controller of the controller
+			root, newRoot = s.resolveController(controller.String())
```

Comment:
should this have a max recursion depth? I think the nuts node does not allow reference depth > 5, so that alone would indicate a problem?

Reply:
It would generate some errors in the nuts node and could possibly be harmful here, but let's fix that when it is encountered. Not worth the effort imho